en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-300-to-310.md

```diff
@@ -0,0 +1,713 @@
+# Upgrading API Manager from 3.1.0 to 3.1.0
+
+The following information describes how to upgrade your API Manager server **from APIM 3.1.0 to 3.1.0**.
+
+!!! note
+    Before you follow this section, see [Upgrading Process](../upgrading-wso2-api-manager/upgrading-process.md) for more information.
+
+!!! attention "Before you Begin"
+    This release is a WUM-only release. This means that there are no manual patches. Any further fixes or latest updates for this release can be updated through the WSO2 Update Manager (WUM).
+
+    -   **If you are upgrading to this version, in order to use this version in your production environment** , use the WSO2 Update Manager and get the latest available updates for WSO2 API Manager 3.1.0. For more information on how to do this, see [Updating WSO2 Products](https://docs.wso2.com/display/updates/Using+WSO2+Update+Manager).
+
+Follow the instructions below to upgrade your WSO2 API Manager server **from WSO2 API-M 3.0.0 to 3.1.0**.
+
+### Preparing for Migration
+#### Disabling versioning in the registry configuration
+
+If there are frequently updating registry properties, having the versioning enabled for registry resources in the registry can lead to unnecessary growth in the registry related tables in the database. To avoid this, versioning has been disabled by default in API Manager 3.1.0.
+
+Therefore, when migrating to API Manager 3.1.0, it is **required** to turn off the registry versioning in your
+current API Manager 3.0.0 version and run the below scripts against **the database that is used by the registry**.
+
+!!! note "NOTE"
+    Alternatively, it is possible to turn on registry versioning in API Manager 3.1.0 and continue. But this is
+    highly **NOT RECOMMENDED** and these configurations should only be changed once.
+    
+!!! warning "Not recommended"
+    If you decide to proceed with registry resource versioning enabled, Add the following configuration to the `<NEW_API-M_HOME>/repository/conf/deployment.toml` file of new WSO2 API Manager. 
+    
+    ```
+    [registry.static_configuration]
+    enable=true
+    ```
+    
+    !!! note "NOTE"
+        Changing these configuration should only be done before the initial API-M Server startup. If changes are done after the initial startup, the registry resource created previously will not be available.
+
+!!! note
+    If you are using WSO2 Identity Server (WSO2 IS) as a Key Manager, follow the instructions in [Upgrading WSO2 IS as the Key Manager to 5.10.0](../upgrading-wso2-is-as-key-manager/upgrading-from-is-km-590-to-5100.md).
+
+-   [Step 1 - Migrate the API Manager configurations](#step-1-migrate-the-api-manager-configurations)
+-   [Step 2 - Upgrade API Manager to 3.1.0](#step-2-upgrade-api-manager-to-300)
```

Comment:
Shall we change the version as 310 here?

Reply:
Fixed with 02d0d1b869f0f90576477aead17e8dfa5c919384